debug/mutex.go

```diff
@@ -0,0 +1,164 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
```

Comment:
Is this mostly copied from the Go sync Lib? If large sections are, do we need to add their copyright here?